mig/install/generateconfs.py

```diff
@@ -42,7 +42,7 @@
 
 # NOTE: __file__ is /MIG_BASE/mig/install/generateconfs.py and we need MIG_BASE
 
-sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+sys.path.append(os.path.realpath(os.path.join(os.path.dirname(__file__), "../..")))
```

Comment:
I'd would prefer to avoid this construct if it's really a matter of  handling `__file__` possibly being a relative path. In that case I'd much prefer to call  `abspath` (or `realpath`) on __file__ first thing. It's only if we e.g. cross symlinks that realpath should be needed...
`sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))`
My brief tests indicate that it will work no matter how `__file__` is executed and across python versions.

Reply:
Ok - for the record this was changed because I was unable to directly execute the script with the previous formulation.

That said, really this dovetails with `PYTHONPATH` and once that is honoured everywhere these all go away. That't not to say it shouldn't be right in-tree, but for the sake of record this is understood as being legacy anyway and there will be a mass removal of straggling stuff like this once it is safe to do so.

Reply:
alright, and yes that sounds like PYTHONPATH and/or abspath might be missing in certain variations of the expansion. If we can just add the `abspath(__file__)` in those situations I'm quite confident we won't introduce _other_ side effects until we can drop the workaround completely.